reactor-core/src/main/java/reactor/core/publisher/MonoLift.java

```diff
@@ -53,4 +53,9 @@ public String stepName() {
 		}
 		return super.stepName();
 	}
+
+	@Override
+	public Object scanUnsafe(Attr key) {
+		return Scannable.from(source).scanUnsafe(key);
```

Comment:
I do not see other `Lift` operators  
are they still in progress?  
Doesn't it ignores `scanUnsafe()` from its super class?  
Does it mean that `CoreSubscriber<? super I> input` considered to be `SYNC`?

Reply:
The `LIFT` operators are still to be done, indeed. 

I'm sorry, I kept the PR opened after we decided to reverse the logic and extend the declaration of the attribute to all operators but I now realize that it's confusing, even with the `WIP` prefix in the title 😞 

But thank you very much for your review, this kind of repetitive task is prone to error 😅  I'll look at other comments once I'm done to make a final "corrective" commit.

Reply:
Fixed in commit [3c4ebed](https://github.com/reactor/reactor-core/pull/2123/commits/3c4ebeda4e237c12428d419b6de2d83a98ac330d)

Reply:
I see that all `Lift` `Publisher`s marked as `SYNC` and It means that lift functions must return `SYNC` `Subscriber`s.  
If it is so - it is better to mention this in `Operators.lift()` etc.  
This behavior makes easier to add more than one `Hooks.onEachOperator(Function)` or  `Hooks.onLastOperator(Function)` as we can assume that `XXXLift` publishers should not be decorated (because of their probably `ASYNC` nature)  
but their sources should be checked and it is hard to understand whether current `SYNC` `Publisher` is `Lift` `Publisher` or not to skip it and traverse to its `source` or just assume that it safe to not decorate it.   
So If we assume that `lifter`s produce only synchronous `CoreSubscriber`s it is better to get `Attr.RUN_STYLE` from its source. So we would not need additional `Attr` or flag `interface` on `Lift` `Publisher`s

Reply:
@robotmrv the lift functions and associated operators are the primary mean through which one can set up a `onEachOperator` hook without running into an infinite loop, aka they don't and shouldn't apply assembly hooks. Perhaps this should be made clearer in the javadoc. But you shouldn't run into the situation of eg. sleuth decorating a `LiftPublisher` (or a subscriber produced by the user-provided lift function).

Reply:
@simonbasle  I think my description of the problem was confusing, so let me explain by example.  
There are two libraries which apply hooks (e.g. `onEachOperator`)  
```java
Mono.just(0)
    .delayElement(Duration.ofMillis(100))//some `ASYNC` operator
//calls onAssembly()
//    apply first LiftFunction -> `MonoLift` (1) decorates as source is `ASYNC` 
//    apply second LiftFunction -> (2) source is `SYNC` so just skips decoration, that is wrong  
    .map(it -> ...)// (3) lib 2 context is lost because it was not propagated 
...
```
at (1) we have `ASYNC` `source` for the first `Lift function` so it decorates `source` `Publisher`  
then at (2) source is `SYNC` so it "thinks" that it is ok to skip decoration  
at (3) context of the second `Lift function` is lost

Reply:
got it, so it should have the "special" behavior of delegating the SYNC/ASYNC to the original subscriber. and if said subscriber is not Core, answer UNKNOWN

Reply:
Fixed in commit [abd4003](https://github.com/reactor/reactor-core/pull/2123/commits/abd40036e30e787e47e07fc7461c90fbdb1102e0)